tests/test-github-policy.sh

```
@@ -475,16 +475,39 @@ echo "==> Checked-in consumer policies equal their derivation"
+if [ -x /bin/bash ] && /bin/bash -c '[ "${BASH_VERSINFO[0]}" -le 4 ]' 2>/dev/null; then
+  diff -q <(/bin/bash "$ROOT/scripts/derive-consumer-policy.sh" touchstone-policy-canary --no-queue) <(bash "$ROOT/scripts/derive-consumer-policy.sh" touchstone-policy-canary --no-queue) >/dev/null \
+    && ok "derivation without --require-status is identical under /bin/bash $(/bin/bash -c 'echo ${BASH_VERSION%%(*}')" \
+    || fail "derivation differs or fails under /bin/bash (bash 3/4)"
+fi
+# --require-status adds exactly one repository-owned context rule and nothing
+# else: the canonical rules are joined, never removed or weakened.
+with_status="$(bash "$ROOT/scripts/derive-consumer-policy.sh" touchstone-policy-canary --no-queue --require-status canary/body-check --require-status canary/body-check)"
+without_status="$(bash "$ROOT/scripts/derive-consumer-policy.sh" touchstone-policy-canary --no-queue)"
+[ "$(jq -c '[.managedRuleset.rules[] | select(.type == "required_status_checks") | .parameters.required_status_checks[].context]' <<<"$with_status")" = '["canary/body-check"]' ] \
+  && [ "$(jq -S 'del(.managedRuleset.rules[] | select(.type == "required_status_checks"))' <<<"$with_status")" = "$(jq -S . <<<"$without_status")" ] \
+  && ok "--require-status adds one deduplicated context rule and changes nothing else" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)